scopes/compositions/compositions/compositions.ui.runtime.tsx

```diff
@@ -1,28 +1,45 @@
 import type { ReactNode } from 'react';
+import { ComponentType } from 'react';
 import { Slot, SlotRegistry } from '@teambit/harmony';
 import { ComponentAspect, ComponentUI } from '@teambit/component';
 import { UIRuntime } from '@teambit/ui';
-
 import { CompositionsSection } from './composition.section';
 import { CompositionsAspect } from './compositions.aspect';
 import { MenuBarWidget } from './compositions';
 
 export type CompositionsMenuSlot = SlotRegistry<MenuBarWidget[]>;
+export type EmptyStateSlot = SlotRegistry<ComponentType>;
 
 export class CompositionsUI {
-  constructor(private menuBarWidgetSlot: CompositionsMenuSlot) {}
+  constructor(private menuBarWidgetSlot: CompositionsMenuSlot, private emptyStateSlot: EmptyStateSlot) {}
+
+  /**
+   * register a new tester empty state. this allows to register a different empty state from each environment for example.
+   */
+  registerEmptyState(emptyStateComponent: ComponentType) {
+    this.emptyStateSlot.register(emptyStateComponent);
+    return this;
+  }
 
   registerMenuWidget(...widget: MenuBarWidget[]) {
     this.menuBarWidgetSlot.register(widget);
   }
 
   static dependencies = [ComponentAspect];
   static runtime = UIRuntime;
-  static slots = [Slot.withType<ReactNode>()];
-
-  static async provider([component]: [ComponentUI], config, [slot]: [CompositionsMenuSlot]) {
-    const compositions = new CompositionsUI(slot);
-    const section = new CompositionsSection(compositions, { menuBarWidgetSlot: compositions.menuBarWidgetSlot });
+  static slots = [Slot.withType<ReactNode>(), Slot.withType<ComponentType>()];
+
+  static async provider(
+    [component]: [ComponentUI],
+    config: {},
+    [compositionMenuSlot, emptyStateSlot]: [CompositionsMenuSlot, EmptyStateSlot]
+  ) {
+    const compositions = new CompositionsUI(compositionMenuSlot, emptyStateSlot);
+    const section = new CompositionsSection(
+      compositions,
+      { menuBarWidgetSlot: compositions.menuBarWidgetSlot },
```

Comment:
move it inside the options (ie. `{ menuBarWidgetSlot: compositions.menuBarWidgetSlot, emptyStateSlot },`

Reply:
this is giving me errors with typescript so leaving this for now. we can make a pr to fix it tomorrow